generate_jlc_files.py

```diff
@@ -29,13 +29,20 @@
 DEFAULT_DB_PATH="cpl_rotations_db.csv"
 
 def main():
-	parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter)
-	parser.add_argument('project_dir', metavar='project directory', type=os.path.abspath, nargs='?', help='Directory of KiCad project', default=os.getcwd())
-	parser.add_argument('-d', '--database', metavar='database', type=str, help='Filename of database', default=os.path.join(os.path.dirname(__file__), DEFAULT_DB_PATH))
-	parser.add_argument('-v', '--verbose', help='increases log verbosity for each occurrence', dest='verbose_count', action="count", default=0)
-	parser.add_argument('--warn-no-lcsc-partnumber', help='warn if ', dest='warn_no_partnumber', action='store_true')
+	parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter, description='Generates BOM and CPL in CSV fashion to be used in JLCPCB Assembly Service', prog='generate_jlc_files',
+                                    usage='%(prog)s [OPTIONS] INPUT_DIRECTORY')
+	parser.add_argument('project_dir', metavar='INPUT_DIRECTORY', type=os.path.abspath, nargs='?', help='Directory of KiCad project', default=os.getcwd())
+	parser.add_argument('-d', '--database', metavar='DATABASE', type=str, help='Filename of database', default=os.path.join(os.path.dirname(__file__), DEFAULT_DB_PATH))
+	verbosity = parser.add_argument_group('verbosity arguments')
+	verbosity.add_argument('-v', '--verbose', help='increases log verbosity for each occurrence', dest='verbose_count', action="count", default=0)
+	verbosity.add_argument('--warn-no-lcsc-partnumber', help='Set warning output if lcsc part number is not found', dest='warn_no_partnumber', action='store_true')
 	parser.add_argument('--assume-same-lcsc-partnumber', help='assume same lcsc partnumber for all components of a group', action='store_true', dest='assume_same_lcsc_partnumber')
-	parser.add_argument('-o', '--output', metavar='output directory', dest='output_dir', type=os.path.abspath, help='Output directory')
+	parser.add_argument('-o', '--output', metavar='OUTPUT_DIRECTORY', dest='output_dir', type=os.path.abspath, help='Output directory. Default: INPUT_DIRECTORY')
+
+	if (len(sys.argv) == 1):
```

Comment:
Why? It has a default argument to use the current working directory.

Reply:
Initially, if someone clones the repo and executes the module without having any KiCad project (which may happen) in the same path, what I mean: 

```shell
python generate_jlc_files.py
```

It will throw at shell:
```
Failed to find netlist file: <dir> (and sub-directories). Run 'Tools -> Generate Bill of Materials' in Eeschema (any format). It will generate an intermediate file we need.
```

This message tells to the user that there is no net list associated, but the root cause is kind of unknown. The reason is that the module is being executed from a directory that does not contain a KiCad project. 

Maybe, the script can be more explicit letting know to the user should use  KiCad Project path. However, I imagine having this module installed using pip as python package in an env environment, in this matter, the workflow would be to always set an input directory (KiCad project) to where to look for files from. Then, it makes sense for me to pop up a message showing to the user a way of how can obtain help and from there get information about the available commands that can be used.

Reply:
Maybe we can improve this error message to point out that
- script either needs to be run within the project directory
- or it needs to be specified as an argument
In addition the directory name also needs to match the project filename. I think some people also stumbled upon this. :)

Reply:
@twam sounds great.

Reply:
I think at this point it's simpler to just take out the default input directory.

I agree with @sophiekovalevsky that the current error message is a bit confusing, and if the user wants to use the current directory,
`python generate_jlc_files.py .`
is only 2 extra characters to type.

This also makes the CLI more "unix-like", which is probably what most people using a command line tool expect.

Reply:
@twam, @matthewlai 

I have made changes based on these discussions. Feel free to give it a try and let me know your thoughts.

In a nutshell: 

- If user does not provide INPUT_DIRECTORY as first argument, display help. 
- User must provide an INPUT_DIRECTORY.
- Suggestion replacement in warn-no-lcsc-partnumber.

Reply:
I prefer the default directory, but can also live with the required input directory, but please then just add `nargs=1` in line 34. This will make argparse require the parameter and print help if used wrongly. You also don't need to overwrite the usage in line 33 then.

Reply:
Why do I need to add nargs=1? 

If an user sets more than one positional argument it will throw the same message as is nargs=1 would not be set. If I adde nargs=1 I will need to convert first INPUT_DIRECTORY into a string and keep the normal flow.

Reply:
You're right. You don't need `nargs=1`. The default is basically the same except that it will not result in a list.

Nevertheless if you don't do the check for `(len(sys.argv) == 1)` then `parser.parse_args` will already detect this and say
```
generate_jlc_files: error: the following arguments are required: INPUT_DIRECTORY
```

I would therefor remove it to keep it simple and leave all the argument parsing to argparse.